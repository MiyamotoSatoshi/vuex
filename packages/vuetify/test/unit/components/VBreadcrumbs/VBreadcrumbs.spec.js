import { test } from '@/test'
import {
  VBreadcrumbs,
  VBreadcrumbsItem
} from '@/components/VBreadcrumbs'
import Vue from 'vue'

test('VBreadcrumbs.js', ({ mount, compileToFunctions }) => {
  it('should have breadcrumbs classes', () => {
    const wrapper = mount(VBreadcrumbs)

    expect(wrapper.hasClass('v-breadcrumbs')).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render items without slot', () => {
    const { render } = compileToFunctions(`
      <v-breadcrumbs :items="items"></v-breadcrumbs>
    `)

    const component = Vue.component('test', {
      components: {
        VBreadcrumbs, VBreadcrumbsItem
      },
      data: () => ({
        items: [
          { text: 'a' },
          { text: 'b' },
          { text: 'c' },
          { text: 'd' }
        ]
      }),
      render
    })
    const wrapper = mount(component)

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should use slot to render items if present', () => {
    const { render } = compileToFunctions(`
      <v-breadcrumbs :items="items">
        <v-breadcrumbs-item slot="item" slot-scope="props" :key="props.item.text">
          {{ props.item.text.toUpperCase() }}
        </v-breadcrumbs-item>
      </v-breadcrumbs>
    `)
    const component = Vue.component('test', {
      components: {
        VBreadcrumbs, VBreadcrumbsItem
      },
      data: () => ({
        items: [
          { text: 'a' },
          { text: 'b' },
          { text: 'c' },
          { text: 'd' }
        ]
      }),
      render
    })
    const wrapper = mount(component)

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should use a custom divider slot', () => {
    const { render } = compileToFunctions(`
      <v-breadcrumbs :items="items">
        <template slot="divider">/divider/</template>
      </v-breadcrumbs>
    `)
    const component = Vue.component('test', {
      components: {
        VBreadcrumbs, VBreadcrumbsItem
      },
      data: () => ({
        items: [
          { text: 'a' },
          { text: 'b' },
          { text: 'c' },
          { text: 'd' }
        ]
      }),
      render
    })
    const wrapper = mount(component)

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should show deprecation notice when using justify-end and justify-start', () => {
    const wrapper = mount(VBreadcrumbs, {
      propsData: {
        justifyEnd: true,
        justifyCenter: true
      },
      slots: {
        default: [{ render: h => h('div') }]
      }
    })

    expect(wrapper.find('.justify-end').length).not.toBe(0)
    expect(wrapper.find('.justify-center').length).not.toBe(0)
    expect(wrapper.html()).toMatchSnapshot()
    expect(`'justify-end' is deprecated, use 'class="justify-end"' instead`).toHaveBeenTipped()
    expect(`'justify-center' is deprecated, use 'class="justify-center"' instead`).toHaveBeenTipped()
  })

  // TODO: this always passes in jest, needs to be e2e
  it.skip('should remove dividers when items change', async () => {
    const component = Vue.component('test', {
      data: () => ({ items: 3 }),
      render (h) {
        return h(VBreadcrumbs, Array.from(Array(this.items), (_, i) => (
          h(VBreadcrumbsItem, { key: i }, [i])
        )))
      }
    })

    const wrapper = mount(component)
    expect(wrapper.html()).toMatchSnapshot()

    wrapper.setData({ items: 1 })
    await wrapper.vm.$nextTick()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
