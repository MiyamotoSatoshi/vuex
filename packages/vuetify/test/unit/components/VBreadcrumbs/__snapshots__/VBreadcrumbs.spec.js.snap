// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VBreadcrumbs.js should have breadcrumbs classes 1`] = `

<ul class="v-breadcrumbs theme--light">
</ul>

`;

exports[`VBreadcrumbs.js should render items without slot 1`] = `

<ul class="v-breadcrumbs theme--light">
  <li>
    <div class="v-breadcrumbs__item">
      a
    </div>
  </li>
  <li class="v-breadcrumbs__divider">
    /
  </li>
  <li>
    <div class="v-breadcrumbs__item">
      b
    </div>
  </li>
  <li class="v-breadcrumbs__divider">
    /
  </li>
  <li>
    <div class="v-breadcrumbs__item">
      c
    </div>
  </li>
  <li class="v-breadcrumbs__divider">
    /
  </li>
  <li>
    <div class="v-breadcrumbs__item">
      d
    </div>
  </li>
</ul>

`;

exports[`VBreadcrumbs.js should show deprecation notice when using justify-end and justify-start 1`] = `

<ul class="v-breadcrumbs justify-center justify-end theme--light">
  <div>
  </div>
</ul>

`;

exports[`VBreadcrumbs.js should use a custom divider slot 1`] = `

<ul class="v-breadcrumbs theme--light">
  <li>
    <div class="v-breadcrumbs__item">
      a
    </div>
  </li>
  <li class="v-breadcrumbs__divider">
    /divider/
  </li>
  <li>
    <div class="v-breadcrumbs__item">
      b
    </div>
  </li>
  <li class="v-breadcrumbs__divider">
    /divider/
  </li>
  <li>
    <div class="v-breadcrumbs__item">
      c
    </div>
  </li>
  <li class="v-breadcrumbs__divider">
    /divider/
  </li>
  <li>
    <div class="v-breadcrumbs__item">
      d
    </div>
  </li>
</ul>

`;

exports[`VBreadcrumbs.js should use slot to render items if present 1`] = `

<ul class="v-breadcrumbs theme--light">
  <li>
    <div class="v-breadcrumbs__item">
      A
    </div>
  </li>
  <li class="v-breadcrumbs__divider">
    /
  </li>
  <li>
    <div class="v-breadcrumbs__item">
      B
    </div>
  </li>
  <li class="v-breadcrumbs__divider">
    /
  </li>
  <li>
    <div class="v-breadcrumbs__item">
      C
    </div>
  </li>
  <li class="v-breadcrumbs__divider">
    /
  </li>
  <li>
    <div class="v-breadcrumbs__item">
      D
    </div>
  </li>
</ul>

`;
